### ISSUE TITLE
4.8.0 update breaks select_advanced field type (multiple = true)

### ISSUE THREAD

**Issue by ryanlabelle** [rilwis/meta-box#771]
After upgrading to 4.8.

Before: http://new.tinygrab.com/6407caa59e5d54225e42221118aaba162e8420551f.png

After: http://new.tinygrab.com/6407caa59e1d50292986c9a10b6ab2bb6e325b81a5.png

Thanks for the urgent response, we are supporting thousands of installs. :)

I originally posted under a recent issue ticket but realized this is a separate issue involving just the core metabox plugin.

**Comment by funkedgeek** [comment]
Can you post the code you are using for select_advanced?  Also, what does it do if you use a regular select?

**Comment by ryanlabelle** [comment]
Sure thing. Thanks.

Our code looks like this:
```
$options = array(
        'id'          => "KEY",
        'name'        =>  'Select individually',
        'type'        => 'post',
        // Post type
        'post_type'   => $post_type,
        // Field type, either 'select' or 'select_advanced' (default)
        'field_type'  => 'select_advanced',
        'multiple'    => true,
        'placeholder' =>  'Select an Item',
        'desc' =>   $desc,
        'query_args'  => array(
            'post_status'    => 'publish',
            'posts_per_page' => - 1,
        ),
        'class' => 'checkbox-condition',
        'tab'  => 'select_list',
    );
```
Regular select seems to work well.

**Comment by ryanlabelle** [comment]
Also, the markup looks close, the posts (select options) are coming through but the select list just isn't working.

Before: http://new.tinygrab.com/6407caa59ecfd5b5a4e347b82617cd78a4bc8f8683.png

After: http://new.tinygrab.com/6407caa59ee2072cb8cfa2e569179905972a9283fa.png

I don't see any JS console errors.

**Comment by funkedgeek** [comment]
Okay, think I fixed it.  The select fields were getting an extra `[]` on the name by accident.  Fixed the culprit.  Download the latest version from github for now, Rilwis will do an update on wordpress.org later tonight.

**Comment by ryanlabelle** [comment]
Thank you!!

On Wed, Feb 17, 2016 at 1:26 PM, Manny Fleurmond <notifications@github.com>
wrote:

> Okay, think I fixed it. The select fields were getting an extra [] on the
> name by accident. Fixed the culprit. Download the latest version from
> github for now, Rilwis will do an update on wordpress.org later tonight.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rilwis/meta-box/issues/771#issuecomment-185412147>.
>



-- 
777 Fort St, Victoria, BC V8W 1H2

Ryan Labelle | Honeycomb Webworks
250-655-9202
<http://twitter.com/honeycombweb>

**Comment by ryanlabelle** [comment]
Sorry bud, still getting a blank box. 

Here is the markup after the recent update:
http://new.tinygrab.com/6407caa59e8498cba0a5931455c313c08915eaad17.png

Here is a screen of the controls:
http://new.tinygrab.com/6407caa59e00474cdc1ee9156ce2a5939e8e7e18bc.png
They are there but when active (mouse click) there is no dropdown

What it normally looks like:
http://new.tinygrab.com/6407caa59e18f169707480f9ebb4976e03c572fc70.png

Let me know how I can help debug.

**Comment by funkedgeek** [comment]
So the select_advanced doesn't work at all?  When you click on it nothing comes up?

**Comment by ryanlabelle** [comment]
That's right. Nothing comes up.

**Comment by ryanlabelle** [comment]
Okay, seems to be an issue with WooCommerce. Perhaps a conflict with chosen?

**Comment by funkedgeek** [comment]
Woocomerce, again......

**Comment by ryanlabelle** [comment]
LOL, I feel your pain, it's a beast! We've added support for it in our latest themes and there are so many plugins to support. Unfortunately WP is now married to it.

**Comment by funkedgeek** [comment]
This is like the third time WooComerce has caused compatibility issues.

**Comment by ryanlabelle** [comment]
How can I help debug? Can I do some testing to assist you?

**Comment by rilwis** [comment]
Looks like WooCommerce uses an older version of Select2 library (3.5.4) and that breaks the JS code of Meta Box (uses 4.x version).

Previously WooCommerce used Chosen library, now Select2. :(

**Comment by rilwis** [comment]
OK, made a fix for that. Works for me. Please test it.

**Comment by ryanlabelle** [comment]
Thank you. I'll test it out tonight. :)

**Comment by ryanlabelle** [comment]
Fixed works great. Thank you very much.
